.github/workflows/zig-xbuild.yml

```diff
@@ -0,0 +1,29 @@
+# This workflow uses Zig and its excellent cross-compilation support to test
+# compiling for multiple platforms. No tests are actually run since it would
+# require emulation.
+# TODO: move from nightly to zig 0.12 once it is released
+name: zig cross-compile
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: ${{ matrix.ttriple }} dll:${{ matrix.shared_libs }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ttriple: [ aarch64-linux-musl, wasm32-wasi, x86_64-linux-gnu.2.27, x86_64-linux-musl, x86_64-windows-gnu ]
```

Comment:
How to cross-compile for MacOS? I've failed to

Reply:
Should be a simple matter of: `zig build -Dtarget=x86_64-macos-none`

Or aarch64-macos-none for apple silicon...

Reply:
This results in an error:
`.../bdwgc/extra/../os_dep.c:4354:10: error: 'mach/exception.h' file not found`

Reply:
E.g. see https://github.com/ivmai/bdwgc/actions/runs/7323413734/job/19945962265

Reply:
@ivmai ah yes, that's correct. I forgot I'm running with a patched zig. Zig doesn't ship with the correct headers just yet. I opened an issue for this https://github.com/ziglang/zig/issues/18257. They're usually quite fast adding headers but I'm guessing it'll be after the holidays.

In Acton we add in a few extra headers, see https://github.com/actonlang/acton/tree/main/deps/zig-extras/lib/libc/include/any-macos-any

You can do the same, just copy into your zig/lib/...

Reply:
Okay, let's wait for right nightly build and then add macos targets.

Reply:
Right yes, agreed, let's wait for it.. I was just thinking if you wanted to test stuff locally first :)